JsonSchema/JsonSchemaExtensions.cs

```diff
@@ -119,10 +119,15 @@ public static JsonSchema Bundle(this JsonSchema jsonSchema, EvaluationOptions? o
 
 				if (schema.Keywords == null) continue;
 
-				schemasToSearch.AddRange(schema.Keywords.OfType<ISchemaContainer>().Select(x => x.Schema));
-				schemasToSearch.AddRange(schema.Keywords.OfType<ISchemaCollector>().SelectMany(x => x.Schemas));
-				schemasToSearch.AddRange(schema.Keywords.OfType<IKeyedSchemaCollector>().SelectMany(x => x.Schemas.Values));
-				schemasToSearch.AddRange(schema.Keywords.OfType<ICustomSchemaCollector>().SelectMany(x => x.Schemas));
+				// I shouldn't ordinarily have to check for null here,
+				// but ItemsKeyword implements two interfaces and only one
+				// will be populated at a time.  There may be others
+				// that implement multiple in the same way, so I need to
+				// consider that.
+				schemasToSearch.AddRange(schema.Keywords.OfType<ISchemaContainer>().Where(x => x.Schema != null!).Select(x => x.Schema));
+				schemasToSearch.AddRange(schema.Keywords.OfType<ISchemaCollector>().Where(x => x.Schemas != null!).SelectMany(x => x.Schemas));
+				schemasToSearch.AddRange(schema.Keywords.OfType<IKeyedSchemaCollector>().Where(x => x.Schemas != null!).SelectMany(x => x.Schemas.Values));
+				schemasToSearch.AddRange(schema.Keywords.OfType<ICustomSchemaCollector>().Where(x => x.Schemas != null!).SelectMany(x => x.Schemas));
```

Comment:
A similar issue can occur in:

https://github.com/gregsdennis/json-everything/blob/44d5fe8f83882d061b336cedc792bcc60563fd3e/JsonSchema/JsonSchema.cs#L324-L350

That's why I suggested having a single place for such an iteration.

Reply:
It can't happen here at least not with `items`.

This switch only runs one branch, and it evaluates them in this order.  If the `items` keyword is a schema, it'll hit the first branch and bypass the second.  If it's an array, it'll bypass the first branch and hit the second.

The code in `.Bundle()` however runs all of the cases.  This was the problem.

Reply:
Okay, so this code is perfectly fine - why not use it as well in `.Bundle()`? Instead of 4x `schemasToSearch.AddRange(schema.Keywords.OfType...)`, there'd be just `schemasToSearch.AddRange(schema.Keywords.SelectMany(JsonSchema.GetSubschemas))` - it just needs to be made `internal` instead of `private`.

Reply:
Interesting.  I'll have a look.

Reply:
10 points to @amis92 for preventing me from re-inventing the wheel.